apps/examples/testcase/le_tc/kernel/tc_task.c

```diff
@@ -414,6 +417,108 @@ static void tc_task_prctl(void)
 	TC_ASSERT_EQ_ERROR_CLEANUP("prctl", strncmp(getname, setname, CONFIG_TASK_NAME_SIZE), 0, get_errno(), prctl(PR_SET_NAME, oldname, 0, 0, 0));
 
 	prctl(PR_SET_NAME, oldname, 0, 0, 0);
+
+#ifdef CONFIG_MESSAGING_IPC
+	const char *portname = "testPort";
+	pid_t pid;
+	int recv_cnt = 0;
+	int *data_arr;
+	struct sched_param param;
+
+	/* saving receiver information */
+	pid = getpid();
+	ret_chk = sched_getparam(pid, &param);
+	TC_ASSERT_EQ("sched_getparam", ret_chk, OK);
+
+	ret_chk = prctl(PR_MSG_SAVE, portname, pid, param.sched_priority);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+	/* Reading receiver information */
+
+	ret_chk = prctl(PR_MSG_READ, getname, data_arr, recv_cnt);
+	TC_ASSERT_NEQ("prctl", data_arr, NULL);
+	TC_ASSERT_NEQ("prctl", recv_cnt, ERROR);
+
+	/* Saving receiver information */
+
+	ret_chk = prctl(PR_MSG_SAVE, portname, pid, param.sched_priority);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+	/* Removing receiver information */
+
+	ret_chk = prctl(PR_MSG_REMOVE, portname);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+#endif
+#ifdef CONFIG_TASK_MONITOR
+	ret_chk = prctl(PR_MONITOR_REGISTER, SEC_2);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+	ret_chk = prctl(PR_MONITOR_UPDATE, NULL);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+#endif
+#ifdef CONFIG_PREFERENCE
+	int test = 0;
+	bool *existing = false;
+	void *value = &test;
+	preference_data_t data;
+
+	data.key = TEST_STRING;
+	data.type = PRIVATE_PREFERENCE;
+	data.attr.type = PREFERENCE_TYPE_INT;
+	data.attr.len = sizeof(int);
+	data.value = value;
+
+	/* Set preference with prctl */
+
+	ret_chk = prctl(PR_SET_PREFERENCE, &data);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+	/* Get preference with prctl */
+
+	ret_chk = prctl(PR_GET_PREFERENCE, &data);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+	data.key = TEST_STRING;
+	data.type = PRIVATE_PREFERENCE;
+	data.attr.type = PREFERENCE_TYPE_BOOL;
+
+	/* Get preference with prctl */
+
+	ret_chk = prctl(PR_GET_PREFERENCE, &data);
+	TC_ASSERT_NEQ("prctl", data.type, ERROR);
+
+	/* Remove preference with prctl */
+
+	ret_chk = prctl(PR_REMOVE_PREFERENCE, PRIVATE_PREFERENCE, TEST_STRING);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+	/* Remove all preference with prctl */
+
+	ret_chk = prctl(PR_REMOVEALL_PREFERENCE, PRIVATE_PREFERENCE, NULL);
+	TC_ASSERT_EQ("prctl", ret_chk, OK);
+
+#endif
+#ifdef CONFIG_SYSTEM_REBOOT_REASON
+	int reboot_reason_test_val = 100;
+
+	/* Read the previous reboot reason */
+
+	ret_chk = prctl(PR_REBOOT_REASON_READ);
+	TC_ASSERT_GEQ("prctl", ret_chk, OK);
```

Comment:
For some cases, read value from reboot reason cannot be OK.
So I think we'd better test like below.
```
1. clear the reboot reason
2. write some value into reboot reason
3. try to read the reboot reason and check that the value is matched to wrote from No.2
```